From e7671c2a3be03d790cbc225cd3e784b5434fb5da Mon Sep 17 00:00:00 2001
From: David Bauer <mail@david-bauer.net>
Date: Mon, 16 Jan 2023 01:29:22 +0100
Subject: [PATCH] config: switch to AC_CHECK_LIB

This fixes spurious build-errors on OpenWrt, where the AM_ICONV macro
is undefined while invoking autoconfig. Later in the build, the ICONV
LDOPTIONS are set to @LIBICONV@, failing the build.

Signed-off-by: David Bauer <mail@david-bauer.net>
---
 configure.ac | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -70,10 +70,7 @@ AC_CHECK_DECLS([getmntent], [], [], [[#i
 AC_CHECK_DECLS([getmntinfo], [], [], [[#include <sys/mount.h>]])
 
 # optional iconv support
-AC_ARG_WITH([iconv], AS_HELP_STRING([--without-iconv], [build without iconv support]))
-if test "x$with_iconv" != "xno"; then
-	AM_ICONV
-fi
+AC_CHECK_LIB(iconv, iconv_open)
 
 # xxd (distributed with vim) is used in the testsuite
 AC_CHECK_PROG([XXD_FOUND], [xxd], [yes])
